src/controllers/auth.js

```diff
@@ -187,4 +193,72 @@ const registerUser = async (req, res) => {
   }
 };
 
-module.exports = { createDefaultTables, updateUser, registerUser };
+const obtainAccessToken = async (req, res) => {
+  // extract payload
+  const { username, password } = req.body.fields;
+
+  try {
+    // check if the username exists in the Db
+    let users = rowService.get({
+      tableName: '_users',
+      whereString: 'WHERE username=?',
+      whereStringValues: [username],
+    });
+
+    if (users.length <= 0) {
+      return res.status(401).send({ message: 'Invalid username or password' });
+    }
+
+    // check if the password is valid
+    let user = users[0];
+    const isMatch = await comparePasswords(password, user.hashed_password);
+
+    if (!isMatch) {
+      return res.status(401).send({ message: 'Invalid username or password' });
+    }
+
```

Comment:
@AbegaM - do we need to document that users can just have one role?  I presume we also need to enforce it (somewhere)